tests/sentry/api/endpoints/test_project_stacktrace_link.py

```diff
@@ -317,3 +318,117 @@ def test_munge_flutter_worked(self, mock_integration):
+            project=self.project,
+            repo=self.repo,
+            stack_root="usr/src/getsentry/",
+            source_root="",
+            automatically_generated=False,
+        )
+        # Created by the user, not as well defined stack root
+        self.code_mapping4 = self.create_code_mapping(
+            organization_integration=self.oi,
+            project=self.project,
+            repo=self.repo,
+            stack_root="usr/src/",
+            source_root="",
+            automatically_generated=False,
+        )
+        # Created by automation, well defined stack root
+        self.code_mapping5 = self.create_code_mapping(
+            organization_integration=self.oi,
+            project=self.project,
+            repo=self.repo,
+            stack_root="usr/src/getsentry/src/sentry/",
+            source_root="",
+            automatically_generated=True,
+        )
+
+        self.filepath = "usr/src/getsentry/src/sentry/src/sentry/utils/safe.py"
+        self.login_as(self.user)
+
+    def expected_configurations(self, code_mapping) -> Mapping[str, Any]:
+        return {
+            "defaultBranch": "master",
+            "id": str(code_mapping.id),
+            "integrationId": str(self.integration.id),
+            "projectId": str(self.project.id),
+            "projectSlug": self.project.slug,
+            "provider": serialized_provider(),
+            "repoId": str(self.repo.id),
+            "repoName": self.repo.name,
+            "sourceRoot": code_mapping.source_root,
+            "stackRoot": code_mapping.stack_root,
+        }
+
+    def test_test_multiple_code_mapping_matches_order(self):
+        project_stacktrace_link_endpoint = ProjectStacktraceLinkEndpoint()
+
+        configs = [
+            self.code_mapping1,
+            self.code_mapping2,
+            self.code_mapping3,
+            self.code_mapping4,
+            self.code_mapping5,
+        ]
+        expected_config_order = [
+            self.code_mapping3,
+            self.code_mapping4,
+            self.code_mapping1,
+            self.code_mapping5,
+            self.code_mapping2,
+        ]
+        sorted_configs = project_stacktrace_link_endpoint.sort_code_mapping_configs(configs)
+        assert sorted_configs == expected_config_order
+
+    def test_multiple_code_mapping_matches(self):
+        sourceUrl = "https://github.com/usr/src/getsentry/src/sentry/src/sentry/utils/safe.py"
+        with mock.patch.object(
+            ExampleIntegration,
+            "get_stacktrace_link",
+            return_value=sourceUrl,
+        ):
+            response = self.get_success_response(
+                self.organization.slug, self.project.slug, qs_params={"file": self.filepath}
+            )
+            # Assert that the best candidate was chosen
+            assert response.data["config"] == self.expected_configurations(self.code_mapping3)
+            assert response.data["sourceUrl"] == sourceUrl
```

Comment:
```suggestion
            assert response.data["sourceUrl"] == "https://github.com/usr/src/getsentry/src/sentry/src/sentry/utils/safe.py"
```

Since sourceUrl is only used once in the whole test.